### PR TITLE
Use SiteExtension.Sdk in Microsoft.AspNetCore.AzureAppServices.SiteExtension

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

**Comment by JunTaoLuo** [comment]
Why are the tests failing?

File: src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj [hunk 1]
```diff
@@ -20,4 +22,15 @@
     <ProjectReference Include="..\Microsoft.Web.Xdt.Extensions\Microsoft.Web.Xdt.Extensions.csproj" PrivateAssets="All" />
   </ItemGroup>
 
+  <ItemGroup>
+    <ProjectReference Include="..\Microsoft.AspNetCore.AzureAppServices.HostingStartup\Microsoft.AspNetCore.AzureAppServices.HostingStartup.csproj" PrivateAssets="All" />
```
Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj]: What is this used for again?

Comment (pakrym) [reply src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj]: To make sure it is packed in the correct order.

**Comment by pakrym** [comment]
It fails because it needs https://github.com/aspnet/BuildTools/pull/515

File: build/dependencies.props [hunk 2]
```diff
@@ -4,6 +4,7 @@
   </PropertyGroup>
   <PropertyGroup Label="Package Versions">
     <InternalAspNetCoreSdkPackageVersion>2.1.0-preview1-15618</InternalAspNetCoreSdkPackageVersion>
+    <InternalAspNetCoreSiteExtensionSdkPackageVersion>2.1.0-preview1-15623</InternalAspNetCoreSiteExtensionSdkPackageVersion>
```
Comment (muratg) [inline build/dependencies.props]: @natemcmaster I assume this is fine but wanted to check.

Comment (natemcmaster) [reply build/dependencies.props]: Yes, this is fine.

**Comment by pakrym** [comment]
@natemcmaster ping

**Pull request merged**